libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/camera/CameraLayoutObserver.kt

```diff
@@ -37,18 +53,45 @@ internal class CameraLayoutObserver(
         binding.coordinatorLayout.removeOnLayoutChangeListener(layoutListener)
     }
 
-    private fun getOverlayEdgeInsets(): EdgeInsets {
-        val bottom = vPadding + (mapView.height - binding.roadNameLayout.top)
+    private fun getPortraitPadding(): EdgeInsets {
+        val top = binding.guidanceLayout.height.toDouble()
+        val bottom = mapView.height.toDouble() - binding.roadNameLayout.top.toDouble()
         return when (store.state.value.navigation) {
             is NavigationState.DestinationPreview,
             is NavigationState.FreeDrive,
             is NavigationState.RoutePreview -> {
-                EdgeInsets(vPadding, hPadding, bottom, hPadding)
+                EdgeInsets(vPadding, hPadding, vPadding.plus(bottom), hPadding)
+            }
+            is NavigationState.ActiveNavigation,
+            is NavigationState.Arrival -> {
+                EdgeInsets(vPadding.plus(top), hPadding, vPadding.plus(bottom), hPadding)
             }
+        }
+    }
+
+    private fun getLandscapePadding(): EdgeInsets {
+        val left = mapView.width.toDouble() - binding.infoPanelLayout.right.toDouble()
+        val right = mapView.width.toDouble() - binding.actionListLayout.left.toDouble()
+        val bottom = mapView.height.toDouble() - binding.roadNameLayout.top.toDouble()
+        return when (store.state.value.navigation) {
+            is NavigationState.FreeDrive -> {
+                EdgeInsets(
+                    vPaddingLandscape,
+                    hPaddingLandscape,
+                    vPaddingLandscape.plus(bottom),
+                    hPaddingLandscape
```

Comment:
Curious if `left` `right` and `bottom` are all zero here. They're already calculated so this block may be unnecessary

Reply:
> left right and bottom

I guess you mean left, right and top

They are not 0 as if you see `vPaddingLandscape` and `hPaddingLandscape` have values. These are the padding values that needs to be applied to the camera. The bottom applies the padding + height of road name label